pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1510,8 +1522,11 @@ func compareCacheWithNodeInfoSnapshot(cache *schedulerCache, snapshot *Snapshot)
 
 	expectedNodeInfoList := make([]*framework.NodeInfo, 0, cache.nodeTree.numNodes)
 	expectedHavePodsWithAffinityNodeInfoList := make([]*framework.NodeInfo, 0, cache.nodeTree.numNodes)
-	for i := 0; i < cache.nodeTree.numNodes; i++ {
-		nodeName := cache.nodeTree.next()
+	nodesList, err := cache.nodeTree.list()
+	if err != nil {
+		return fmt.Errorf("Unexpected error : %v", err.Error())
```

Comment:
Fatal for this as well pls.

Reply:
This is fmt, it is supposed to return an error.

Reply:
You can pass `testing.T` as an argument

Reply:
done, thank you